lib/ffmpeg/transcoder.rb

```diff
@@ -29,19 +29,16 @@ def initialize(movie, output_file, options = EncodingOptions.new, transcoder_opt
       @output_file = output_file
       @transcoder_options = transcoder_options
 
+      self.ensure_temp_directory
+
       # If the movie has frames with varying resolutions, we need to pre-encode the movie for trims
       # This is because ffmpeg can't reliably handle inputs that contain frames with differing resolutions particularly for trimming with `filter_complex`
       @movie.check_frame_resolutions if @transcoder_options[:permit_dynamic_resolution_pre_encode]
 
       if requires_pre_encode
         @movie.paths.each do |path|
           # Make the interim path folder if it doesn't exist
-          dirname = "#{File.dirname(path)}/interim"
-          unless File.directory?(dirname)
-            FileUtils.mkdir_p(dirname)
-          end
-
-          interim_path = "#{File.dirname(path)}/interim/#{File.basename(path, File.extname(path))}_#{SecureRandom.urlsafe_base64}.mp4"
+          interim_path = "/tmp/interim/#{File.basename(path, File.extname(path))}_#{SecureRandom.urlsafe_base64}.mp4"
```

Comment:
mega nit: I see the new path being used in a few different places. Can we use a constant for the `/tmp/interim`?

Reply:
Good call out on this, yeah, probably ideal to make `/tmp/rlovelett/` a global of some sort